File: Digital_Image_Processing/exp_1/exp_1_5.py
# -*- coding:utf-8 -*- 

import matplotlib.pyplot as plt
import Image
import matplotlib.cm as cm
import numpy as np

fig = plt.figure()
circuit_0 = np.asarray(Image.open('circuit.tif').convert('L'))/float(255)
ax1 = fig.add_subplot(221)
ax1.imshow(circuit_0,cmap=cm.gray)
ax1.set_title('input image')
r = np.arange(0,1+0.0001,0.0001)
s = 1 - r
ax3 = fig.add_subplot(2,2,(3,4))
ax3.plot(r,s)
ax3.set_xlabel('r')
ax3.set_ylabel('s')
ax3.set_title('s-r')

circuit_1 = circuit_0.copy()
(M,N) = circuit_0.shape
for m in range(M):
    for n in range(N):
        circuit_1[m,n] = 1 - circuit_0[m,n]
ax2 = fig.add_subplot(222)
ax2.imshow(circuit_1,cmap=cm.gray)
ax2.set_title('output image')
fig.suptitle('exp_1_5')
fig.savefig('exp_1_5.jpg')

File: Digital_Image_Processing/matlab_ref/exp_1/exp_1_5.m
cd d:/
cd DIP
cd exp_1
circuit_0 = imread('d:/DIP/images/circuit.tif');
circuit_0 = im2double(circuit_0);
[M,N] = size(circuit_0);


subplot(1,3,1);
imshow(circuit_0);
title('source image');

r = 0:0.0001:1;
s = 1-r;
subplot(1,3,2);
plot(r,s);
xlabel('r');
ylabel('s');
title('s-r');

circuit_1 = 1 - circuit_0;
subplot(1,3,3);
imshow(circuit_1);
title('transformed image');
imwrite(circuit_1,'circuit_1.jpg');